client/snapshot.go

```diff
@@ -0,0 +1,151 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/strutil"
+)
+
+// errors on reading:
+var (
+	ErrSnapshotUnsupported  = errors.New("unsupported snapshot format")
```

Comment:
It's better to not introduce these global variables in the client until we have a strong reason for them. Even more in this package, which is more of a public API, and once here we cannot go back and cannot change the error handling. In practice, though, the code generally doesn't care about which of these N errors happened. Another side effect of this approach is that we lose the ability to provide good error messages. For example, which format is unsupported in the message above? How was the size wrong in the message below?  These are simpler and most probably good enough as fmt.Errorf for now. Then, if someday we do find a reason to distinguish individual errors, we can add the exact thing we need (or someone else) without breaking anyone.